src/main/java/bbd/jportal2/Main.java

```diff
@@ -1,145 +1,50 @@
-
-    @Parameter(names = {"--builtin-postprocessor", "-bpp"}, description = "Built-In (Java-based) generatorName to run. Format is <generator_name>:<dest_dir> i.e. --builtin-postprocessor=CSNetCode:./cs")
-    private List<String> builtinPostProcessors = new ArrayList<>();
-
-    @Parameter(names = {"--template-postprocessor", "-tpp"},
-            description =
-                    "FreeMarker-based post-processor to run."
-                            + "Format is <free_marker_generator_name>:<dest_dir> i.e. "
-                            + "'--template-postprocessor=MyCustomGenerator:./output'. "
-                            + "The template must exist as a directory under the location specified by --template-location.")
-    private List<String> templatePostProcessors = new ArrayList<>();
-
-    @Parameter(names = {"--template-location", "-tl"}, description = "Freemarker template location. Default is <current_working_directory>/jportal2_templates")
-    private List<String> templateLocations = Arrays.asList(Paths.get(System.getProperty("user.dir"), "jportal2_templates").toString());
-
-    @Parameter(names = { "--flag", "-F"}, description = "Flags to pass to the generator")
-    private List<String> flags = new ArrayList<>();
-
-    @Parameter(names = { "--help", "-h", "-?" }, help = true)
-    private boolean help;    
-    //===============================================================================================  
 
+public class Main {
 
     private static final Logger logger = LoggerFactory.getLogger(Main.class);
-    private BufferedReader bufferedReader;
-
 
     /**
      * Reads input from stored repository
      */
-    public static void main(String args[])
-    {
-        Main main = new Main();
-
-        try 
-        {
-            JCommander jCommander = JCommander.newBuilder()
-                                        .addObject(main)
-                                        .build();
-
-            jCommander.setProgramName("JPortal2");
-            jCommander.parse(args);
+    public static void main(String[] args) {
+        JPortal2Arguments arguments = JPortal2ArgumentParser.parse(args);
+        //Set this before the logger starts.
+        if (arguments.getLogFileName() != null)
+            System.setProperty("log.name", arguments.getLogFileName());
 
-            if (main.help || args.length == 0) {
-                jCommander.usage();
-                return;
-            }            
-        }
-        catch (ParameterException exc)
-        {
-            logger.error("Error", exc);
-            exc.getJCommander().usage();
-            System.exit(1);
-        }
+        try {
+            ProjectCompiler pj = ProjectCompilerBuilder.build(arguments);
 
-        //Set this before the logger starts.
-        if (main.logFileName != null)
-            System.setProperty("log.name", main.logFileName);
+            if (Objects.nonNull(pj)) {
+                int rc = pj.compileAll();
```

Comment:
Don’t think this if should be nested?

Reply:
The `if` looks nested but the `if` above it has been moved. There's no curly brace, it's a standalone `if`.

Reply:
Ok I see. Please add braces, fix whitespace or write on one line?